bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java

```diff
@@ -7074,15 +7097,11 @@ protected boolean containsSavedState(IMemento memento) {
 	 * @since 3.3
 	 */
 	protected void doRestoreState(IMemento memento) {
-		Integer offset= memento.getInteger(TAG_SELECTION_OFFSET);
-		if (offset == null)
+		ISelection savedSelection = readSelectionFromMemento(memento);
```

Comment:
A common pattern across other selection-using methods of AbstractTextEditor is checking with `getSelectionProvider().isValidSelection(selection)` - it might be useful to add that check inside of this code?

Reply:
If it wasn't done earlier, don't bother adding it now. Maybe in a future PR.

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/issues/1320
Documented here. Feel free to close if you feel like that Issue is too nitpicky